src/radical/pilot/bootstrapper/default_bootstrapper.sh

```diff
@@ -1426,10 +1426,27 @@ then
     FINAL_SLEEP=30
     echo "# -------------------------------------------------------------------"
     echo "#"
-    echo "# Sleeping for $FINAL_SLEEP seconds to let dust settle ..."
+    echo "# We wait for at most 30 seconds for the FS to flush profiles."
+    echo "# Success is assumed when all profiles end with a 'QED' event."
     echo "#"
     echo "# -------------------------------------------------------------------"
-    sleep $FINAL_SLEEP
+    nprofs=`echo *.prof | wc -w`
+    nqed=`tail -n 1 *.prof | grep QED | wc -l`
+    nsleep=0
+    while ! test "$nprofs" = "$nqed"
```

Comment:
Hmmm, wait. I might have been misinterpreting that. Let me look a bit closer later.